src/EnergyPlus/DXCoils.cc

```diff
@@ -15720,7 +15719,7 @@ void GetDXCoilIndex(EnergyPlusData &state,
 }
 
 std::string
-GetDXCoilName(EnergyPlusData &state, int &DXCoilIndex, bool &ErrorsFound, Optional_string_const ThisObjectType, Optional_bool_const SuppressWarning)
+GetDXCoilName(EnergyPlusData &state, int &DXCoilIndex, bool &ErrorsFound, std::string_view const ThisObjectType, bool const SuppressWarning)
```

Comment:
Functions like this should not be necessary. This should just be `state.dataDXCoils->DXCoils(DXCoilNum).Name`. The error checking and defaulting should not be necessary. You should not need to call a function to grab a field of an object whose index is known. I understand that this has nothing to do with this PR, but this is as good a place as any for me to hang this comment.

Reply:
These functions should hopefully only get called during the input phase, but (a) good point and (b) I agree.